session/index.js

```diff
@@ -0,0 +1,171 @@
+       * Session timeout: By default, a session lasts until there's 30 minutes of inactivity,
+       * but you can configure this limit using "timeout" load option
+       */
+      if (
+        options &&
+        options.sessions &&
+        typeof options.sessions.timeout === "number"
+      ) {
+        const { timeout } = options.sessions;
+        // In case user provides 0 as the timeout, auto session tracking will be disabled
+        if (timeout === 0) {
+          logger.warn(
+            '[Session]:: Provided timeout value 0 will disable the auto session tracking feature.',
+          );
+          this.sessionInfo.autoTrack = false;
+        }
+        // In case user provides a setTimeout value greater than 0 but less than 10 seconds SDK will show a warning
+        // and will proceed with it
+        if (timeout > 0 && timeout < MIN_SESSION_TIMEOUT) {
+          logger.warn('[Session]:: It is not advised to set "timeout" less than 10 seconds');
+        }
+        this.timeout = timeout;
+      }
+      // If auto session tracking is enabled start the session tracking
+      if (this.sessionInfo.autoTrack) {
+        this.startAutoTracking();
+      } else if (
+        Object.keys(this.sessionInfo).length > 1 &&
+        !this.sessionInfo.manuallyTrackSession
+      ) {
+        /**
+         * Use case:
+         * By default user session is enabled which means storage will have session data.
+         * In case user wanted to opt out and set auto track to false through load option,
+         * clear stored session info.
+         */
+        this.end();
+      }
+    } catch (e) {
+      handleError(e);
+    }
+  }
+
+  /**
+   * A function to validate current session and return true/false depending on that
+   * @param {number} timestamp
+   * @returns boolean
+   */
+  isValidSession(timestamp) {
+    return timestamp <= this.sessionInfo.expiresAt;
+  }
+
+  /**
+   * A function to check for existing session details and depending on that create a new session.
+   */
+  startAutoTracking() {
+    const timestamp = Date.now();
+    if (!this.isValidSession(timestamp)) {
+      this.sessionInfo.id = timestamp.toString(); // set the current timestamp
+      this.sessionInfo.expiresAt = timestamp + this.timeout; // set the expiry time of the session
+      this.sessionInfo.sessionStart = true;
+      this.sessionInfo.autoTrack = true;
+    }
+    this.storage.setSessionInfo(this.sessionInfo);
+  }
+
+  /**
+   * A public method to start a session
+   * @param {string} sessionId     session identifier
+   * @returns
+   */
+  start(sessionId) {
+    if (this.sessionInfo.manuallyTrackSession) return;
+    if (sessionId && typeof sessionId !== 'string') {
+      logger.error(`[Session]:: "sessionId" should be in string format`);
+      return;
+    }
+    this.sessionInfo = {
+      id: sessionId || Date.now().toString(),
+      sessionStart: true,
+      manuallyTrackSession: true,
+      autoTrack: false,
```

Comment:
This is not required as you are resetting the whole object.

Reply:
What is not required?

Reply:
`autoTrack: false`

Reply:
Addressed